TRLevelControlTests/Base/TestBase.cs

```diff
@@ -0,0 +1,263 @@
+
+        // Test compression against original
+        CollectionAssert.AreEqual(new byte[] { 0xCD, 0xCD, 0xCD, 0xCD, 0xCD, 0xCD }, level2.LevelDataChunk.Seperator);
+        CollectionAssert.AreEqual(TRZlib.Decompress(level.Texture32Chunk.CompressedChunk), TRZlib.Decompress(level2.Texture32Chunk.CompressedChunk));
+        CollectionAssert.AreEqual(TRZlib.Decompress(level.Texture16Chunk.CompressedChunk), TRZlib.Decompress(level2.Texture16Chunk.CompressedChunk));
+        CollectionAssert.AreEqual(TRZlib.Decompress(level.SkyAndFont32Chunk.CompressedChunk), TRZlib.Decompress(level2.SkyAndFont32Chunk.CompressedChunk));
+    }
+
+    public static TRLevel WriteReadTempLevel(TRLevel level)
+        => WriteReadTempLevel(level, "TEST1.PHD");
+
+    public static TR2Level WriteReadTempLevel(TR2Level level)
+        => WriteReadTempLevel(level, "TEST1.TR2");
+
+    public static TR3Level WriteReadTempLevel(TR3Level level)
+        => WriteReadTempLevel(level, "TEST1.TR2");
+
+    public static TR4Level WriteReadTempLevel(TR4Level level)
+        => WriteReadTempLevel(level, "TEST1.TR4");
+
+    public static TR5Level WriteReadTempLevel(TR5Level level)
+        => WriteReadTempLevel(level, "TEST1.TRC");
+
+    public static TRLevel WriteReadTempLevel(TRLevel level, string levelName)
+    {
+        string path = GetWritePath(levelName, TRGameVersion.TR1);
+        TR1LevelReader reader = new();
+        TR1LevelWriter writer = new();
+        writer.WriteLevelToFile(level, path);
+        return reader.ReadLevel(path);
+    }
+
+    public static TR2Level WriteReadTempLevel(TR2Level level, string levelName)
+    {
+        string path = GetWritePath(levelName, TRGameVersion.TR2);
+        TR2LevelReader reader = new();
+        TR2LevelWriter writer = new();
+        writer.WriteLevelToFile(level, path);
+        return reader.ReadLevel(path);
+    }
+
+    public static TR3Level WriteReadTempLevel(TR3Level level, string levelName)
+    {
+        string path = GetWritePath(levelName, TRGameVersion.TR3);
+        TR3LevelReader reader = new();
+        TR3LevelWriter writer = new();
+        writer.WriteLevelToFile(level, path);
+        return reader.ReadLevel(path);
+    }
+
+    public static TR4Level WriteReadTempLevel(TR4Level level, string levelName)
+    {
+        string path = GetWritePath(levelName, TRGameVersion.TR4);
+        TR4LevelReader reader = new();
+        TR4LevelWriter writer = new();
+        writer.WriteLevelToFile(level, path);
+        return reader.ReadLevel(path);
+    }
+
+    public static TR5Level WriteReadTempLevel(TR5Level level, string levelName)
+    {
+        string path = GetWritePath(levelName, TRGameVersion.TR5);
+        TR5LevelReader reader = new();
+        TR5LevelWriter writer = new();
+        writer.WriteLevelToFile(level, path);
+        return reader.ReadLevel(path);
+    }
+}
```

Comment:
Can we use in-memory file buffers for this rather than temporary files?

Reply:
I've added a comment to remind myself to add support for this in the updated IO library - agree it will be a better approach for testing and indeed may be useful elsewhere.